File: sample-videochat-webrtc/sample-videochat-webrtc/Components/Call/Views/ParticipantsView/ParticipantVideoView/ParticipantVideoView.m
//
//  ParticipantVideoView.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 15.08.2021.
//  Copyright © 2021 QuickBlox Team. All rights reserved.
//

#import "ParticipantVideoView.h"
#import "Profile.h"

@interface ParticipantVideoView()
//MARK: - IBOutlets
@property (weak, nonatomic) IBOutlet UILabel *userNameTopLabel;
@end

@implementation ParticipantVideoView
//MARK - Setup
- (void)setupViews {
    [super setupViews];
    
    self.userNameTopLabel.hidden = YES;
}

- (void)setVideoView:(UIView *)videoView {
    _videoView = videoView;
    
    self.userNameTopLabel.hidden = NO;
    [self.videoContainerView insertSubview:videoView atIndex:0];
    self.videoContainerView.hidden = NO;
    videoView.translatesAutoresizingMaskIntoConstraints = NO;
    [videoView.leftAnchor constraintEqualToAnchor:self.videoContainerView.leftAnchor].active = YES;
    [videoView.rightAnchor constraintEqualToAnchor:self.videoContainerView.rightAnchor].active = YES;
    [videoView.topAnchor constraintEqualToAnchor:self.videoContainerView.topAnchor].active = YES;
    [videoView.bottomAnchor constraintEqualToAnchor:self.videoContainerView.bottomAnchor].active = YES;
    [videoView layoutIfNeeded];
}

- (void)setName:(NSString *)name {
    [super setName:name];
    
    self.userNameTopLabel.text = self.name;
}

- (void)setParticipantId:(NSNumber *)userID {
    [super setParticipantId:userID];
    Profile *profile = [[Profile alloc] init];
    if (profile.ID == userID.unsignedIntValue) {
        self.userNameTopLabel.text = @"You";
    }
}

- (void)setupHiddenViews {
    self.callingInfoLabelHeightConstraint.constant = 0.0f;
    self.stateLabel.hidden = NO;
    self.videoContainerView.hidden = YES;
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/Calls/Views/ParticipantView/ParticipantVideoView.swift
//
//  ParticipantVideoView.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 19.08.2021.
//  Copyright © 2021 QuickBlox. All rights reserved.
//

import UIKit
import QuickbloxWebRTC

class ParticipantVideoView: ParticipantView {
    
    //MARK: - IBOutlets
    @IBOutlet weak var userNameTopLabel: UILabel!
    @IBOutlet weak var videoContainerView: UIView!
    
    
    //MARK: - Properties
    var videoView: UIView? {
        didSet {
            guard let view = videoView else {
                return
            }
            userNameTopLabel.isHidden = false
            videoContainerView.insertSubview(view, at: 0)
            videoContainerView.isHidden = false
            view.translatesAutoresizingMaskIntoConstraints = false
            view.leftAnchor.constraint(equalTo: videoContainerView.leftAnchor).isActive = true
            view.rightAnchor.constraint(equalTo: videoContainerView.rightAnchor).isActive = true
            view.topAnchor.constraint(equalTo: videoContainerView.topAnchor).isActive = true
            view.bottomAnchor.constraint(equalTo: videoContainerView.bottomAnchor).isActive = true
            view.layoutIfNeeded()
        }
    }
    
    override var name: String {
        didSet {
            userNameTopLabel.text = name
            nameLabel.text = name
            userAvatarLabel.text = String(name.capitalized.first ?? Character("Q"))
        }
    }
    
    override var ID: UInt {
        didSet {
            if Profile().ID == ID {
                userNameTopLabel.text = "You"
            }
            userAvatarLabel.backgroundColor = ID.generateColor()
        }
    }
    
    override func setupViews() {
        super.setupViews()
        
        userNameTopLabel.isHidden = true
    }
    
    override func setupHiddenViews() {
        callingInfoLabelHeightConstraint.constant = 0.0
        stateLabel.isHidden = false
        videoContainerView.isHidden = true
    }
}
